scripts/api.js

```diff
@@ -2,62 +2,85 @@
  * @description This module communicates with Lychee's API
  */
 
-let api = {
-	onError: null,
-};
+/**
+ * @callback APISuccessCB
+ * @param {Object} data the decoded JSON response
+ * @returns {void}
+ */
 
-api.isTimeout = function (errorThrown, jqXHR) {
```

Comment:
So how do you handle the time out error? :thinking:

Reply:
The method was a complete misnomer. First of all note that this method does _not handle_ a timeout error, but simply tries to determine, _whether_ the error is a timeout error. (The method is called `isTimeout`.) Moreover, the logic to determine a timeout was wicked, unreliable and nonsense. That is why I call it a misnomer.

Also note that the only usage of this method was to distinguish, whether the web front-end shows one out of two possible error messages. This is the old code where the method was used:

```js
api.onError(api.isTimeout(errorThrown, jqXHR) ? "Session timed out." : "Server error or API not found.", params, errorThrown);
```

This code is completely gone. The front-end simply shows the error message which has been reported by the backend as part of the exception. This will give the user a much more useful error message than those two generic messages which also were complete nonsense: neither did the server time-out if there has been a response (this is a contradiction within itself) nor was the API not found (otherwise the request would have failed with error code 404).

Reply:
I re-added the code in https://github.com/LycheeOrg/Lychee-front/pull/282/commits/66e4914b80cef861c662a7ee291fab308ea85749

But only after I committed and pushed the fix, I realized that it only works in debug mode. In productive mode the back trace is not serialized (due to security concerns) and hence the condition

```js
lycheeException.previous_exception.exception === "Illuminate\\Session\\TokenMismatchException"
```

will fail in the frontend. However, this is my general idea. I only need to add code to the backend that a `TokenMismatchException` is not wrapped into a generic exception when not in productive mode.

Reply:
I hopefully found and fixed all cases in: https://github.com/LycheeOrg/Lychee-front/pull/282/commits/36313b940732f7fc06c476261310116a835e6df9 and https://github.com/LycheeOrg/Lychee/pull/1217/commits/e7397b3996ae30143b02147d1001ad1d390d81ae